### DISCUSSION TITLE
queryCache.refetchQueries behaviour

### DISCUSSION THREAD

**Discussion by otaciliolacerda** [TanStack/query#596] [General] [answered]
I have the following use case:

- A page that shows a specific item by ID. 
- There is only one ID being shown at any given moment.
- The query key looks like `['my_query_key', id]`. 
- Several mutations in the code might trigger a refetch of this query. 

Now suppose I open the page with ID 1 and then later I check the page for ID 2, 3, 4, 5. When I access the next ID the query for the previous become inactive and is marked for garbage collection in the state. So at this moment I have 5 queries: 1 active and 4 waiting for the period to be garbage collected. At this point if I do a mutation and refetch the query using only the prefix `my_query_key` without the ID (I am trying to avoid the props waterfall) the refetch triggers 5 requests.

I've passed a custom function to `queryKeyOrPredicateFn` to check for the `queryCache.state.markForGarbageCollection` flag. Something like:
```javascript
queryCache.refetchQueries(query => {
        return query.queryKey[0] === QUERY.MEDIA_SET && !query.state.markedForGarbageCollection;
      });
```

Can I trust this flag? 
Is there a reason to refetch queries waiting to be garbage collected?

**Comment by tannerlinsley** [comment, marked as answer]
This is a bug. It’s also fixed in 2.0 ;) go try it out! Info in the releases section.

**Reply by otaciliolacerda** [reply]
I will do. Thank you.
